src/shogun/machine/gp/Inference.cpp

```diff
@@ -214,105 +207,47 @@ get_negative_log_marginal_likelihood_derivatives(CMap<TParameter*, CSGObject*>*
 	// create lock object
 	CLock lock;
 
-#ifdef HAVE_PTHREAD
-	if (num_deriv<2)
+	#pragma omp parallel for
+	for (index_t i=0; i<num_deriv; i++)
 	{
-#endif /* HAVE_PTHREAD */
-		for (index_t i=0; i<num_deriv; i++)
-		{
-			CMapNode<TParameter*, CSGObject*>* node=params->get_node_ptr(i);
-
-			GRADIENT_THREAD_PARAM thread_params;
+        CMapNode<TParameter*, CSGObject*>* node=params->get_node_ptr(i);
+        SGVector<float64_t> gradient;
+        CLock* locked = &lock;
```

Comment:
does this really need to be a pointer? you are not using lock as a function arg anymore anywhere afaik